segmentstore/server/src/main/java/io/pravega/segmentstore/server/writer/StorageWriter.java

```diff
@@ -519,4 +537,103 @@ public String toString() {
     }
 
     //endregion
+
+    //region ProcessorCollection
+
+    /**
+     * Wraps a collection of WriterSegmentProcessors, including the main Segment Aggregator.
+     */
+    private static class ProcessorCollection implements WriterSegmentProcessor {
+        private final SegmentAggregator aggregator;
+        private final List<WriterSegmentProcessor> processors;
+
+        ProcessorCollection(SegmentAggregator aggregator, Collection<WriterSegmentProcessor> processors) {
```

Comment:
I don't understand why you need to separate out the aggregator. Is it convenience to avoid having to look for it in the collection? Since this is internal, I don't really care, but I'd like to at least have a clear justification as a response to this comment, please.

Reply:
Yes, it's a convenience so I don't have to always access the first element in the list when I need it.